src/theory/uf/theory_uf.cpp

```diff
@@ -296,7 +296,7 @@ Node TheoryUF::getNextDecisionRequest( unsigned& priority ){
   }
 }
 
-void TheoryUF::explain(TNode literal, std::vector<TNode>& assumptions, eq::EqProof* pf) {
+void TheoryUF::explain(TNode literal, std::vector<TNode>& assumptions, std::shared_ptr<eq::EqProof> pf) {
```

Comment:
It seems like `std::shared_ptr<EqProof> eqp` should be a `std::weak_ptr` since `explain` is not taking (shared) ownership.